modules/porous_flow/src/bcs/PorousFlowSinkBC.C

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowSinkBC.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowSinkBC);
+
+InputParameters
+PorousFlowSinkBC::validParams()
+{
+  InputParameters params = MooseObject::validParams();
+  params.addRequiredParam<std::vector<BoundaryName>>(
+      "boundary", "The list of boundary IDs from the mesh where this boundary condition applies");
+  params.addRequiredCoupledVar("pressure", "Pressure");
+  params.addRequiredParam<UserObjectName>(
+      "PorousFlowDictator", "The UserObject that holds the list of PorousFlow variable names");
+  params.addRequiredParam<UserObjectName>("fp", "The name of the user object for fluid properties");
+  params.addRequiredParam<Real>("T_in", "Specified inlet temperature");
+  params.addRequiredParam<FunctionName>(
+      "flux_function",
+      "The flux.  The flux is OUT of the medium: hence positive values of "
+      "this function means this BC will act as a SINK, while negative values "
+      "indicate this flux will be a SOURCE. However, this BC only makes physical sense if "
+      "flux_function <= 0.  This function is measured in kg.m^-2.s^-1.");
```

Comment:
maybe make a common InputParams to share between all the classes.  Eliminates typing the same thing in more than once and makes it easier for further modifications.